hathor/p2p/sync_checkpoints.py

```diff
@@ -184,14 +185,26 @@ def run_sync(self):
 
         self._is_running = True
         try:
-            if self.peers_to_request:
-                self.peer_syncing = self.peers_to_request[0]
+            peer_to_request = self.get_peer_to_request()
+            if peer_to_request:
+                self.peer_syncing = peer_to_request
                 self._run_sync()
             else:
                 self.log.debug('no peers to sync from, try again later')
         finally:
             self._is_running = False
 
+    def get_peer_to_request(self) -> Optional['NodeBlockSync']:
+        """
+        """
+        # XXX: we could use a better peer selecting strategy here
+        for peer in self.peers_to_request[:]:
+            if peer.protocol.state is None:
```

Comment:
I can do that. I used the check `state is None` because that's where I got some errors from, in practice. But it's equivalent, state can only ever be none after `connectionLost` is called.

Reply:
Actually I prefer to keep it as `peer.protocol.state`, at least for now. We use this check on sync-v1 too, and we don't have a simple method for `is_connected`, which could be implemented by checking if `protocol.transport is None` or if `protocol.transport` has lost the connection, however this can be transport dependent (in practice we use TCP transport, but in tests we use a mock transport) and there are some nuances with that that'd need to be addressed.